applications/GeoMechanicsApplication/python_scripts/geomechanics_solvers_wrapper.py

```diff
@@ -30,6 +35,10 @@ def CreateSolver(model, custom_settings):
         raise Exception(err_msg)
 
     module_full_name = 'KratosMultiphysics.GeoMechanicsApplication.' + solver_module_name
-    solver = import_module(module_full_name).CreateSolver(model, custom_settings["solver_settings"])
-
+    solver = import_module(module_full_name).CreateSolver(model, params)
     return solver
+
+
+def CreateSolver(model, custom_settings):
+    parallelism =  custom_settings["problem_data"]["parallel_type"].GetString()
+    return CreateSolverByParameters(model, custom_settings,parallelism)
```

Comment:
@avdg81 and I discussed this part and we believe that if we pass the custom_settings["solver_settings"] here instead of custom_settings, the if-statement in line 12-16 is not needed.
Since in this function we assume anyway we're in the 'top-level' of the parameters (since we are able to get the "problem_data" part out in line 43), we should be able to make the following change
```suggestion
    return CreateSolverByParameters(model, custom_settings["solver_settings"], parallelism)
```